### DISCUSSION TITLE
Email in Flex - Agent experience

### DISCUSSION THREAD

**Discussion by nicedisguise** [twilio-labs/paste#1289] [Office hours] [answered]
The specific questions I have are...

- The agent UI has been designed on the UDL and uses several Paste components. I would like the team to review the screens and point out anything that is potentially off.
- We would like the team's thoughts on the end-to end flow of email and to feed back of the UX and patters used.

Background: Email as a Flex channel is going to be released to private beta in August. It will be built on the new guidelines for the UI (UDL and Paste).
Stage of design: Exploration
Research: ongoing
Links to mockups or prototype: 
[Scenario 1 - Inbound Email](https://twilio.invisionapp.com/share/G310F9YPD94N#/446762673_Email_In_Flex_Scenario_01)
[Scenario 2 - Outbound Email ](https://twilio.invisionapp.com/share/MS10F9Z79VDU#/446764382_Email_In_Flex_Scenario_02)
[Scenario 3 - Email threading](https://twilio.invisionapp.com/share/M710H7I2Y49T#/447306529_Email_In_Flex_Scenario_04)

**Comment by TheSisb** [comment]
Hi @nicedisguise,

Thanks for sharing these mockups. We'll chat about it internally in preparation for the office hours. Do you have any areas in particular you would like us to focus on? We want to make sure we cover the most important bits in our 20minute slot.

Thanks

**Reply by nicedisguise** [reply]
Generally, I'd like to take the time to go over the Agent UI and gather your thoughts as to the application of the UDL theme into Flex. In addition, the focus is on the email flows that we are currently testing with users, would be great to get feedback on some of the components used and some of the Flex-specific variants

**Comment by TheSisb** [comment]
Hi @nicedisguise,

I had a chance to chat with the team some more about the accessibility of the outbound email customer directory screen. First, what you have now is totally fine; since the Select button is more of a visual cue, and you can click on the entire row, then there's no concern.  However if a future designer were to add a second button to the row, then it wouldn't be fine for accessibility.  We brainstormed some future-safe alternatives, taking into consideration Modals were not a path forward for you.

The first idea is to treat the directory list similar to [items in the combobox](http://localhost:8000/components/combobox#examples) in terms of keyboard nav and highlight color: 
![image](https://user-images.githubusercontent.com/1592327/111836197-4e1a8980-88cc-11eb-99a4-a871a2cd2a87.png)


Then, we wondered if the whole experience could be made into a combobox, because there was already a question about if the Tab wrapper was necessary at all. You can already have the second combobox below it for selecting a channel, or you can show it upon selecting a user as progressive disclosure. This may allow you to simplify the flow into a single page.

We stopped there because we don't have more context on the product, but ultimately came away with the feeling that the directory seemed a lot like a combobox :)

Hope this helps.

**Comment by nkrantz** [comment, marked as answer]
Link to Office Hours recording can be found here: https://www.loom.com/share/5febc29a470541c1ab0e153257085262 (pw is in #help-design-system slack)
